test/livebook/live_markdown/markdown_helpers_test.exs

```diff
@@ -98,6 +148,38 @@ defmodule Livebook.LiveMarkdown.MarkdownHelpersTest do
       assert markdown == reformat(markdown)
     end
 
+    test "alternate headings" do
+      markdown = """
+      Heading 1
+      =========
+
+      Heading 2
+      ---------
+
+      Body\
+      """
+
+      assert markdown == reformat(markdown)
+    end
+
+    test "manual headings" do
+      markdown = """
+      <h1>Heading 1</h1>
+
+      <h2>Heading 2</h2>
+
+      <h3>Heading 3</h3>
+
+      <h4>Heading 4</h4>
+
+      <h5>Heading 5</h5>
+
+      <h6>Heading 6</h6>\
+      """
+
+      assert markdown == reformat(markdown)
```

Comment:
Included because I didn't expect this result - my assumption was that `<h3>Foo</h3>` would be formatted back to `### Foo`, not passed along mostly-verbatim.

Reply:
Such markdown is parsed as verbatim and could contain children subtree, so we keeping it as is. Could have a special case if the context is just a single binary though.